wallet-new/src/Cardano/Wallet/Kernel.hs

```diff
@@ -203,11 +202,8 @@ applyBlock :: PassiveWallet
 applyBlock pw@PassiveWallet{..} b
     = do
         blocksByAccount <- prefilterBlock' pw b
-        -- TODO(@uroboros/ryan) do proper metadata initialisation (as part of CBR-239: Support history tracking and queries)
-        let blockMeta = BlockMeta . InDb $ Map.empty
-
         -- apply block to all Accounts in all Wallets
-        void $ update' _wallets $ ApplyBlock (blocksByAccount, blockMeta)
+        void $ update' _wallets $ ApplyBlock blocksByAccount
```

Comment:
Yup, like this structure better (having the block metadata part of the prefiltered block). The spec stipulates that, too :)